src/ui/pages/WalletSelect/WalletSelect.tsx

```diff
@@ -75,21 +80,34 @@ function PortfolioRow({ walletGroups }: { walletGroups: WalletGroup[] }) {
   );
 }
 
+const ZERION_ORIGIN = 'https://app.zerion.io';
+
 export function WalletSelect() {
   const navigate = useNavigate();
-  const { data: walletGroups, isLoading } = useQuery({
+  const { data: walletGroups, isLoading: isLoadingWalletGroups } = useQuery({
     queryKey: ['wallet/uiGetWalletGroups'],
     queryFn: () => walletPort.request('uiGetWalletGroups'),
     useErrorBoundary: true,
   });
-
-  const ownedAddressesCount = useMemo(
+  const ownedGroups = useMemo(
     () =>
-      (walletGroups || [])
-        .filter((group) => !isReadonlyContainer(group.walletContainer))
-        .reduce((sum, group) => sum + group.walletContainer.wallets.length, 0),
+      walletGroups?.filter(
+        (group) => !isReadonlyContainer(group.walletContainer)
+      ),
     [walletGroups]
   );
+  const ownedAddresses = useMemo(
+    () =>
+      ownedGroups?.flatMap((group) =>
+        group.walletContainer.wallets.map((wallet) => wallet.address)
+      ) || [],
+    [ownedGroups]
+  );
+
+  const { data: walletsMeta, isLoading: isLoadingWalletsMeta } =
+    useWalletsMetaByChunks({ addresses: ownedAddresses });
+
+  const ownedAddressesCount = ownedAddresses.length ?? 0;
```

Comment:
how can `length` be nullish?

Reply:
ah, it used to be like `ownedAddresses?.length ?? 0` a few minutes ago... that's why, updated ✅